src/thread.cpp

```diff
@@ -158,10 +178,39 @@ void zmq::thread_t::setSchedulingParameters(int priority_, int schedulingPolicy_
 #endif
 
     posix_assert (rc);
+
+    if (use_nice_instead_priority &&
+            thread_priority != ZMQ_THREAD_PRIORITY_DFLT)
+    {
+        // assume the user wants to decrease the thread's nice value
+        // i.e., increase the chance of this thread being scheduled: try setting that to
+        // maximum priority.
+        rc = nice(-20);
+
+        // IMPORTANT: ignore return code: EPERM is typically returned for unprivileged processes
```

Comment:
This should be an assert - if a user sets the priority but it cannot be done due to lack of privileges, it should not be ignored silently as it's a programming error.